git_machete/client.py

```diff
@@ -709,6 +709,9 @@ def traverse(
 
             needs_slide_out: bool = self.__is_merged_to_upstream(
                 branch, opt_no_detect_squash_merges=opt_no_detect_squash_merges)
+            if branch in self.annotations:
+                if self.annotations[branch].qualifiers.slide_out is not None:
+                    needs_slide_out = self.annotations[branch].qualifiers.slide_out
```

Comment:
Hmmm but the current logic can set `needs_slide_out` to `True` even if it was `False` originally, right? 🤔 the qualifiers should only ever cause do the opposite change: to `False` even if it was `True`

Reply:
No, the qualifiers can only force the change from `True` to `False`. To explain the logic you are referring to, the value of `self.annotations[branch].qualifiers.slide_out` can only be `None` if the qualifier is not present in which case the value of `needs_slide_out` remains unchanged. The value of `self.annotations[branch].qualifiers.slide_out` can be `False` if the qualifier is present and equals exactly to `slide-out=no` and in that case it would change the value of `needs_slide_out` to `False` whether it was `True` or `False` beforehand.

Reply:
Hmmm still... pls note that the current approach introduces an inconsistency, compared with how `needs_parent_sync`/`qualifiers.rebase` and `needs_remote_sync`/`qualifiers.push` is handled 🤔 

Neither `qualifiers.push` nor `qualifiers.rebase` can ever be `None`, only `False` and then:

```python
                if needs_parent_sync and branch in self.annotations:
                    needs_parent_sync = self.annotations[branch].qualifiers.rebase
```